src/FlaUILibrary/flaui/util/treeitems.py

```diff
@@ -1,139 +1,155 @@
+
+        return node_count
+
+    @staticmethod
+    def expand_all_tree_nodes(nodes: Any):
         """
-        This function selects a tree item with the given name in tree
-        if item not found a flauierror will be thrown.
+        Expand all tree nodes.
+
+        Args:
+            nodes (Object): TreeItems[] from current node
         """
-        for item in self.treeitems:
-            state = item.ExpandCollapseState
-            self.current_treeitem = item
-            if self.current_treeitem.Name == name:
-                self.current_treeitem.Select()
-                self.treeitems_name.append(self.current_treeitem.Name)
-            #The element is Expanded All children are visible.
-            if state == 1:
-                self.treeitems = item.Items
-                self.select_visible_treeitem_by_name(name)
-            #The element is PartiallyExpanded Some, but not all, children are visible..
-            elif state == 2:
-                self.treeitems = item.Items
-                self.select_visible_treeitem_by_name(name)
-            #The element is Collapsed No children are visible.
-            elif state == 0:
-                continue
-            #The element is LeafNode does not expand or collapse.
-            elif state == 3:
-                continue
-        if not self.treeitems_name:
-            raise FlaUiError(FlaUiError.ElementNameNotFound.format(name))
-        return True
+        for node in nodes:
+            if node.ExpandCollapseState in (ExpandCollapseState.Expanded, ExpandCollapseState.Collapsed):
+                node.Expand()
+                TreeItems.expand_all_tree_nodes(node.Items)
 
-    def expand_all_treeitems(self):
+    @staticmethod
+    def collapse(nodes: Any):
         """
-        This function expands every expandable tree item in tree
+        Collapses every collapsable tree item in root level.
+
+        Args:
+            nodes (Object): TreeItems[] from current node
         """
-        for item in self.treeitems:
-            state = item.ExpandCollapseState
-            self.current_treeitem = item
-            #The element is PartiallyExpanded Some, but not all, children are visible..
-            if state == 2:
-                self.current_treeitem.Expand()
-                self.treeitems = item.Items
-                self.expand_all_treeitems()
-            #The element is Collapsed No children are visible.
-            elif state == 0:
-                self.current_treeitem.Expand()
-                self.treeitems = item.Items
-                self.expand_all_treeitems()
-            #The element is LeafNode does not expand or collapse.
-            elif state == 3:
-                continue
-            #The element is Expanded All children are visible.
-            elif state == 1:
-                continue
-        return True
-
-    def execute_by_location(self, location: Any, exec_func: Any):
```

Comment:
I am not sure if this will really cover all the cases. Because as you see it was a recursive call. Specially every node could still have children nodes which are not expanded so within the newer implementation those are not covered. Or i am seeing it wrong